rebasehelper/tags.py

```diff
@@ -80,7 +80,52 @@ def _map_sections_to_parsed(cls, raw_content: SpecContent, parsed_content: SpecC
 
     @classmethod
     def _parse(cls, raw_content: SpecContent, parsed_content: SpecContent) -> Tuple[Tag]:
-        """Parses all tags from provided SPEC content and determines if they are valid.
+        result = []
+        parsed_mapping = cls._map_sections_to_parsed(raw_content, parsed_content)
+        for section_index, (section, section_content) in enumerate(raw_content.sections):
+            section = section.lower()
+            parsed_index = parsed_mapping[section_index]
+            parsed: List[str] = [] if parsed_index == -1 else parsed_content.sections[parsed_index][1]
+            if section.startswith('%package'):
+                result.extend(cls._parse_package_tags(section, section_content, parsed))
+            elif section.startswith('%sourcelist') or section.startswith('%patchlist'):
+                result.extend(cls._parse_list_tags(section, section_content, parsed, cast(Tuple[Tag], tuple(result))))
+        return cast(Tuple[Tag], tuple(result))
+
+    @classmethod
+    def _parse_list_tags(cls, section: str, section_content: List[str], parsed: List[str],
+                         parsed_tags: Tuple[Tag]) -> List[Tag]:
+        """Parses all tags from a %sourcelist or %patchlist section.
+
+        Only parses tags that are valid (AKA are in parsed), nothing more can
```

Comment:
I don't think AKA is the proper term here, `(that is - are in parsed)` would look better.